// randomness-js/src/generated/factories/IVRFV2PlusWrapper__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  IVRFV2PlusWrapper,
  IVRFV2PlusWrapperInterface,
} from "../IVRFV2PlusWrapper";

const _abi = [
  {
    type: "function",
    name: "calculateRequestPriceNative",
    inputs: [
      {
        name: "_callbackGasLimit",
        type: "uint32",
        internalType: "uint32",
      },
      {
        name: "_numWords",
        type: "uint32",
        internalType: "uint32",
      },
    ],
    outputs: [
      {
        name: "",
        type: "uint256",
        internalType: "uint256",
      },
    ],
    stateMutability: "view",
  },
  {
    type: "function",
    name: "estimateRequestPriceNative",
    inputs: [
      {
        name: "_callbackGasLimit",
        type: "uint32",
        internalType: "uint32",
      },
      {
        name: "_numWords",
        type: "uint32",
        internalType: "uint32",
      },
      {
        name: "_requestGasPriceWei",
        type: "uint256",
        internalType: "uint256",
      },
    ],
    outputs: [
      {
        name: "",
        type: "uint256",
        internalType: "uint256",
      },
    ],
    stateMutability: "view",
  },
  {
    type: "function",
    name: "lastRequestId",
    inputs: [],
    outputs: [
      {
        name: "",
        type: "uint256",
        internalType: "uint256",
      },
    ],
    stateMutability: "view",
  },
  {
    type: "function",
    name: "requestRandomWordsInNative",
    inputs: [
      {
        name: "_callbackGasLimit",
        type: "uint32",
        internalType: "uint32",
      },
      {
        name: "_requestConfirmations",
        type: "uint16",
        internalType: "uint16",
      },
      {
        name: "_numWords",
        type: "uint32",
        internalType: "uint32",
      },
      {
        name: "extraArgs",
        type: "bytes",
        internalType: "bytes",
      },
    ],
    outputs: [
      {
        name: "requestId",
        type: "uint256",
        internalType: "uint256",
      },
    ],
    stateMutability: "payable",
  },
] as const;

export class IVRFV2PlusWrapper__factory {
  static readonly abi = _abi;
  static createInterface(): IVRFV2PlusWrapperInterface {
    return new Interface(_abi) as IVRFV2PlusWrapperInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): IVRFV2PlusWrapper {
    return new Contract(address, _abi, runner) as unknown as IVRFV2PlusWrapper;
  }
}
