// randomness-js/src/generated/factories/IVRFMigratableConsumerV2Plus__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  IVRFMigratableConsumerV2Plus,
  IVRFMigratableConsumerV2PlusInterface,
} from "../IVRFMigratableConsumerV2Plus";

const _abi = [
  {
    type: "function",
    name: "setCoordinator",
    inputs: [
      {
        name: "vrfCoordinator",
        type: "address",
        internalType: "address",
      },
    ],
    outputs: [],
    stateMutability: "nonpayable",
  },
  {
    type: "event",
    name: "CoordinatorSet",
    inputs: [
      {
        name: "vrfCoordinator",
        type: "address",
        indexed: false,
        internalType: "address",
      },
    ],
    anonymous: false,
  },
] as const;

export class IVRFMigratableConsumerV2Plus__factory {
  static readonly abi = _abi;
  static createInterface(): IVRFMigratableConsumerV2PlusInterface {
    return new Interface(_abi) as IVRFMigratableConsumerV2PlusInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): IVRFMigratableConsumerV2Plus {
    return new Contract(
      address,
      _abi,
      runner
    ) as unknown as IVRFMigratableConsumerV2Plus;
  }
}
