// randomness-js/src/generated/factories/VRFV2PlusWrapperConsumerBase__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Interface, type ContractRunner } from "ethers";
import type {
  VRFV2PlusWrapperConsumerBase,
  VRFV2PlusWrapperConsumerBaseInterface,
} from "../VRFV2PlusWrapperConsumerBase";

const _abi = [
  {
    type: "function",
    name: "getBalance",
    inputs: [],
    outputs: [
      {
        name: "",
        type: "uint256",
        internalType: "uint256",
      },
    ],
    stateMutability: "view",
  },
  {
    type: "function",
    name: "getLinkToken",
    inputs: [],
    outputs: [
      {
        name: "",
        type: "address",
        internalType: "contract LinkTokenInterface",
      },
    ],
    stateMutability: "view",
  },
  {
    type: "function",
    name: "i_vrfV2PlusWrapper",
    inputs: [],
    outputs: [
      {
        name: "",
        type: "address",
        internalType: "contract IVRFV2PlusWrapper",
      },
    ],
    stateMutability: "view",
  },
  {
    type: "function",
    name: "rawFulfillRandomWords",
    inputs: [
      {
        name: "_requestId",
        type: "uint256",
        internalType: "uint256",
      },
      {
        name: "_randomWords",
        type: "uint256[]",
        internalType: "uint256[]",
      },
    ],
    outputs: [],
    stateMutability: "nonpayable",
  },
  {
    type: "error",
    name: "OnlyVRFWrapperCanFulfill",
    inputs: [
      {
        name: "have",
        type: "address",
        internalType: "address",
      },
      {
        name: "want",
        type: "address",
        internalType: "address",
      },
    ],
  },
] as const;

export class VRFV2PlusWrapperConsumerBase__factory {
  static readonly abi = _abi;
  static createInterface(): VRFV2PlusWrapperConsumerBaseInterface {
    return new Interface(_abi) as VRFV2PlusWrapperConsumerBaseInterface;
  }
  static connect(
    address: string,
    runner?: ContractRunner | null
  ): VRFV2PlusWrapperConsumerBase {
    return new Contract(
      address,
      _abi,
      runner
    ) as unknown as VRFV2PlusWrapperConsumerBase;
  }
}
