// randomness-js/src/generated/IVRFV2PlusWrapper.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import type {
  BaseContract,
  BigNumberish,
  BytesLike,
  FunctionFragment,
  Result,
  Interface,
  ContractRunner,
  ContractMethod,
  Listener,
} from "ethers";
import type {
  TypedContractEvent,
  TypedDeferredTopicFilter,
  TypedEventLog,
  TypedListener,
  TypedContractMethod,
} from "./common";

export interface IVRFV2PlusWrapperInterface extends Interface {
  getFunction(
    nameOrSignature:
      | "calculateRequestPriceNative"
      | "estimateRequestPriceNative"
      | "lastRequestId"
      | "requestRandomWordsInNative"
  ): FunctionFragment;

  encodeFunctionData(
    functionFragment: "calculateRequestPriceNative",
    values: [BigNumberish, BigNumberish]
  ): string;
  encodeFunctionData(
    functionFragment: "estimateRequestPriceNative",
    values: [BigNumberish, BigNumberish, BigNumberish]
  ): string;
  encodeFunctionData(
    functionFragment: "lastRequestId",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "requestRandomWordsInNative",
    values: [BigNumberish, BigNumberish, BigNumberish, BytesLike]
  ): string;

  decodeFunctionResult(
    functionFragment: "calculateRequestPriceNative",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "estimateRequestPriceNative",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "lastRequestId",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "requestRandomWordsInNative",
    data: BytesLike
  ): Result;
}

export interface IVRFV2PlusWrapper extends BaseContract {
  connect(runner?: ContractRunner | null): IVRFV2PlusWrapper;
  waitForDeployment(): Promise<this>;

  interface: IVRFV2PlusWrapperInterface;

  queryFilter<TCEvent extends TypedContractEvent>(
    event: TCEvent,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TypedEventLog<TCEvent>>>;
  queryFilter<TCEvent extends TypedContractEvent>(
    filter: TypedDeferredTopicFilter<TCEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TypedEventLog<TCEvent>>>;

  on<TCEvent extends TypedContractEvent>(
    event: TCEvent,
    listener: TypedListener<TCEvent>
  ): Promise<this>;
  on<TCEvent extends TypedContractEvent>(
    filter: TypedDeferredTopicFilter<TCEvent>,
    listener: TypedListener<TCEvent>
  ): Promise<this>;

  once<TCEvent extends TypedContractEvent>(
    event: TCEvent,
    listener: TypedListener<TCEvent>
  ): Promise<this>;
  once<TCEvent extends TypedContractEvent>(
    filter: TypedDeferredTopicFilter<TCEvent>,
    listener: TypedListener<TCEvent>
  ): Promise<this>;

  listeners<TCEvent extends TypedContractEvent>(
    event: TCEvent
  ): Promise<Array<TypedListener<TCEvent>>>;
  listeners(eventName?: string): Promise<Array<Listener>>;
  removeAllListeners<TCEvent extends TypedContractEvent>(
    event?: TCEvent
  ): Promise<this>;

  calculateRequestPriceNative: TypedContractMethod<
    [_callbackGasLimit: BigNumberish, _numWords: BigNumberish],
    [bigint],
    "view"
  >;

  estimateRequestPriceNative: TypedContractMethod<
    [
      _callbackGasLimit: BigNumberish,
      _numWords: BigNumberish,
      _requestGasPriceWei: BigNumberish
    ],
    [bigint],
    "view"
  >;

  lastRequestId: TypedContractMethod<[], [bigint], "view">;

  requestRandomWordsInNative: TypedContractMethod<
    [
      _callbackGasLimit: BigNumberish,
      _requestConfirmations: BigNumberish,
      _numWords: BigNumberish,
      extraArgs: BytesLike
    ],
    [bigint],
    "payable"
  >;

  getFunction<T extends ContractMethod = ContractMethod>(
    key: string | FunctionFragment
  ): T;

  getFunction(
    nameOrSignature: "calculateRequestPriceNative"
  ): TypedContractMethod<
    [_callbackGasLimit: BigNumberish, _numWords: BigNumberish],
    [bigint],
    "view"
  >;
  getFunction(
    nameOrSignature: "estimateRequestPriceNative"
  ): TypedContractMethod<
    [
      _callbackGasLimit: BigNumberish,
      _numWords: BigNumberish,
      _requestGasPriceWei: BigNumberish
    ],
    [bigint],
    "view"
  >;
  getFunction(
    nameOrSignature: "lastRequestId"
  ): TypedContractMethod<[], [bigint], "view">;
  getFunction(
    nameOrSignature: "requestRandomWordsInNative"
  ): TypedContractMethod<
    [
      _callbackGasLimit: BigNumberish,
      _requestConfirmations: BigNumberish,
      _numWords: BigNumberish,
      extraArgs: BytesLike
    ],
    [bigint],
    "payable"
  >;

  filters: {};
}
